src/components/actionSheet/index.tsx

```diff
@@ -0,0 +1,265 @@
+   * function)
+   */
+  options?: Array<ButtonProps>;
+  /**
+   * callback for when dismissing the action sheet, usually used for setting visible prop to false
+   */
+  onDismiss?: DialogProps['onDismiss'];
+  /**
+   * Should use the native action sheet for iOS
+   */
+  useNativeIOS?: boolean;
+  /**
+   * When passed (only with useNativeIOS), will display a cancel button at the bottom (overrides cancelButtonIndex)
+   */
+  showCancelButton?: boolean;
+  /**
+   * Add or override style of the action sheet (wraps the title and actions)
+   */
+  containerStyle?: StyleProp<ViewStyle>;
+  /**
+   * Add or override style of the dialog wrapping the action sheet
+   */
+  dialogStyle?: StyleProp<ViewStyle>;
+  /**
+   * Add or override style of the options list
+   */
+  optionsStyle?: StyleProp<ViewStyle>;
+  /**
+   * Render custom title
+   */
+  renderTitle?: () => JSX.Element | JSX.Element[];
+  /**
+   * Render custom action
+   * Note: you will need to call onOptionPress so the option's onPress will be called
+   */
+  renderAction?: (
+    option: ButtonProps,
+    index: number,
+    onOptionPress: ActionSheetOnOptionPress
+  ) => JSX.Element | JSX.Element[];
+  /**
+   * Called once the modal has been dismissed (iOS only, modal only)
+   */
+  onModalDismissed?: DialogProps['onModalDismissed'];
+  /**
+   * Whether or not to handle SafeArea
+   */
+  useSafeArea?: boolean;
+  /**
+   * testID for e2e tests
+   */
+  testID?: string;
+};
+
+/**
+ * @description: Cross platform Action Sheet, with a support for native iOS solution
+ * @gif: https://media.giphy.com/media/l0HUpXOR6RqB2ct5S/giphy.gif
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/ActionSheetScreen.js
+ */
+class ActionSheet extends Component<ActionSheetProps, {}> {
+  static displayName = 'ActionSheet';
+
+  constructor(props: ActionSheetProps) {
+    super(props);
+
+    this.onOptionPress = this.onOptionPress.bind(this);
+    this.renderAction = this.renderAction.bind(this);
+  }
+
+  static defaultProps = {
+    title: undefined,
+    message: undefined,
+    useModal: false
+  };
+
+  componentDidUpdate(prevProps: ActionSheetProps) {
+    const {useNativeIOS} = this.props;
+    const wasVisible = prevProps.visible;
+    const willBeVisible = this.props.visible;
+
+    if (!wasVisible && willBeVisible && useNativeIOS && Constants.isIOS) {
+      const {title, message, cancelButtonIndex, destructiveButtonIndex, options, showCancelButton} = this.props;
+
+      const optionsArray = options !== undefined ? options : [];
+      let cancelBtnIndex = cancelButtonIndex;
+      if (showCancelButton) {
+        optionsArray.push({label: 'Cancel'});
+        cancelBtnIndex = optionsArray.length - 1;
+      }
+
+      ActionSheetIOS.showActionSheetWithOptions(
+        {
+          title,
+          message,
+          options: _.flatMap(optionsArray, 'label'),
```

Comment:
This was `_.map` before, was this intentional?

Reply:
Yes, `_.map` can return nulls elements inside an array, which we don't support, nor with the type we declared.
(`_.flatMap` will return array with defined values or an empty array)

Reply:
On second thought, why would anyone send an `option` with `label=null`?

Reply:
Fixed, thank you